### PR TITLE
Docker dev env

### PR DESCRIPTION
**Pull request by wiese**

### CODE REVIEW

File: build/app/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
+FROM php:7.1-fpm
+
+RUN mkdir -p /usr/share/nginx/www/spenden.wikimedia.de/current/var/cache
+RUN mkdir -p /usr/share/nginx/www/spenden.wikimedia.de/current/var/doctrine_proxies
+RUN mkdir -p /usr/share/nginx/www/spenden.wikimedia.de/current/var/log
+RUN chown -R www-data:www-data /usr/share/nginx/www/spenden.wikimedia.de/current/var
```
[the remark this answers is not in the document]
Comment (wiese) [reply build/app/Dockerfile]: Developed step by step and not groomed thereafter - this way [clearly against the best practice](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#minimize-the-number-of-layers).

**Comment by gbirke** [comment]
Rebased on current master. @timEulitz will try it out on his machine and when he's successful, we can  merge this.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      - mailhog
+
+  database:
+    image: mysql:5.6
+    environment:
+      - MYSQL_RANDOM_ROOT_PASSWORD=yes
+      - MYSQL_USER=fundraising
+      - MYSQL_PASSWORD=INSECURE PASSWORD
+      - MYSQL_DATABASE=fundraising
+    expose:
```
Comment (gbirke) [inline docker-compose.yml]: @wiese Is there a reason why there is no volume defined for the database? The current setup will delete all data and tables as soon as docker-compose is stopped.

Comment (wiese) [reply docker-compose.yml]: No explicit reason but I did not consider it needed so far. The container (and its volume) will be reused unless you explicitly remove it.
Mind "Where to Store Data" in https://hub.docker.com/_/mysql/

**Comment by JeroenDeDauw** [comment]
image

**Comment by JeroenDeDauw** [comment]
Please beware that the README has a table of contents at its top which is manually maintained. So when changing the section headers or removing/adding/moving sections, this thing needs updating. This PR broke the TOC without anyone noticing, hence me mentioning it here, since perhaps no one was aware of the need to keep this in mind.

**Comment by gbirke** [comment]
Thank for pointing this out, I've added an npm task to autogenerate the TOC in #1221

**Pull request merged**